### PR TITLE
++ healthchecks to pg&redis

### LINKED ISSUE #104: Update Docker Compose file to improve service startup dependencies
Upon working with the current `docker-compose.yml`, I have noticed that some services are starting before their dependencies are ready. This leads to initial startup failures and makes the docker-compose process less reliable.

Specifically, I observed that services depending on the Redis instances and PostgreSQL databases were starting before these dependencies were fully initialized. Additionally, the PostgreSQL services (`txs-db` and `cfg-db`) did not have health checks, which is necessary for reliably determining when these services are ready to accept connections.

To address these problems, I propose the following changes:

1. Add health checks to the `txs-db` and `cfg-db` services, using the `pg_isready -U postgres` command to determine if the PostgreSQL servers are accepting connections.
2. Add the `service_healthy` condition to all services that depend on `txs-db` or `cfg-db`, to ensure they only start after the databases are ready.
3. Add a health check to the `txs-redis` and `cgw-redis` services, using the `redis-cli ping` command to determine if the Redis server is up.
4. Add the `service_healthy` condition to all services that depend on `txs-redis` and `cgw-redis`, to ensure they only start after Redis is ready.

### PR DESCRIPTION
**Pull request by mshakeg**
Fixes #104

### CODE REVIEW

**Comment by mshakeg** [comment]
[Contributor License Agreement](https://safe.global/cla/) returns Page Not Found

**Comment by Uxio0** [comment]
> [Contributor License Agreement](https://safe.global/cla/) returns Page Not Found

Noted, there was a typo on the url! Thanks! https://github.com/safe-global/safe-infrastructure/pull/108

File: docker-compose.yml [hunk 1]
```diff
@@ -26,18 +26,32 @@ services:
...
+      retries: 3
 
   cfg-db:
     image: postgres:14-alpine
     environment:
       POSTGRES_PASSWORD: postgres
     volumes:
       - ./data/cfg-db:/var/lib/postgresql/data
-
+    healthcheck:
```
Comment (Uxio0) [inline docker-compose.yml:41]: Could you inherit using `<<:` yaml so you don't have to define the same liveness multiple times for redis and postgres?

**Review by Uxio0 — CHANGES_REQUESTED** [review]
Thank you very much for PR!

**Comment by mshakeg** [comment]
I have read the CLA Document and I hereby sign the CLA

recheck

**Comment by mshakeg** [comment]
recheck

**Review by Uxio0 — APPROVED** [review]
Really nice, thanks!

**Pull request merged**